gdx/test/com/badlogic/gdx/math/QuaternionTest.java

```diff
@@ -0,0 +1,23 @@
+
+package com.badlogic.gdx.math;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class QuaternionTest {
+    private static final float epsilon = Float.MIN_NORMAL;
+
+    @Test
+    public void testEulerPrecision () {
+        float yaw = -5f;
+        float pitch = 42.00055f;
+        float roll = 164.32f;
```

Comment:
There should probably be tests on more than just these 3 inputs, especially if you're using as strict an epsilon as `Float.MIN_NORMAL`. I also think this test isn't really doing the same thing as what happens in your original GIFs, where multiple rotations are applied, one after the next, to check for loss of accuracy. Performing a sequence of rotations that we can know, mathematically, should end with the same angles as the start would be the easiest to verify.

Reply:
My test here was the bare minimum to exercise the case that round-tripping an Euler value introduces a high degree of imprecision. I do like the suggestion of adding an additional test to illustrate the case of multiple rotations that should result in a known value.